### PR TITLE
SDT-40: Update existing Judgement and Warrant Request combined workflow

### PR DESCRIPTION
**Pull request by kiran-yenigala-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SDT-40

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

**Review by markdathornehmcts — APPROVED** [review]
LGTM

File: consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/api/CMCApi.java [hunk 1]
```diff
@@ -64,4 +66,12 @@ WarrantResponse warrantRequest(@RequestHeader(AUTHORIZATION) String authorizatio
                                    @RequestHeader(SDT_REQUEST_ID) String sdtRequestId,
                                    @RequestBody WarrantRequest warrantRequest);
 
+    @PostMapping("/requestWarrant")
```
Comment (paulridout) [inline consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/api/CMCApi.java:69]: Should the path be "/requestJudgmentAndWarrant"?

File: consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/api/IJudgementWarrantService.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package uk.gov.moj.sdt.cmc.consumers.api;
+
+import uk.gov.moj.sdt.cmc.consumers.request.JudgementWarrantRequest;
+import uk.gov.moj.sdt.cmc.consumers.request.WarrantRequest;
+import uk.gov.moj.sdt.cmc.consumers.response.JudgementWarrantResponse;
+import uk.gov.moj.sdt.cmc.consumers.response.WarrantResponse;
```
Comment (paulridout) [inline consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/api/IJudgementWarrantService.java:6]: Are the WarrantRequest and WarrantResponse imports needed?

File: consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/client/impl/JudgementWarrantService.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    private CMCApi cmcApi;
+
+    @Autowired
+    public JudgementWarrantService(CMCApi cmcApi) {
+        this.cmcApi = cmcApi;
+    }
+
+    @Override
+    public JudgementWarrantResponse judgementWarrantRequest(String authorization,
+                                                            String serviceAuthorization,
```
Comment (paulridout) [inline consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/client/impl/JudgementWarrantService.java:22]: These two parameters aren't on the corresponding method on the other service classes (e.g. JudgementRequestService.requestJudgment()) - are they needed?

File: consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/client/mock/MockJudgementWarrantService.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+    @Override
+    public JudgementWarrantResponse judgementWarrantRequest(String authorization,
+                                                            String serviceAuthorization,
+                                                            String idamId,
+                                                            String sdtRequestId,
+                                                            JudgementWarrantRequest judgementWarrantRequest) {
+        JudgementWarrantResponse response = new JudgementWarrantResponse();
+        response.setWarrantNumber("123456");
+        response.setFee(45678L);
```
Comment (paulridout) [inline consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/client/mock/MockJudgementWarrantService.java:19]: Might be good to set at least one of the judgment related fields as well, just to distinguish it from a warrant.

File: consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/client/impl/JudgementWarrantService.java (hunk 3 above)
Comment (kiran-yenigala-hmcts) [reply consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/client/impl/JudgementWarrantService.java:22]: Better to have them, service authorisation is needed to communicate with CMC later part

File: consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/api/CMCApi.java (hunk 1 above)
Comment (kiran-yenigala-hmcts) [reply consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/api/CMCApi.java:69]: changed

File: consumers/src/unit-test/java/uk/gov/moj/sdt/cmc/consumers/client/mock/MockJudgementWarrantServiceTest.java [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    @Test
+    void getClient() {
+        JudgementWarrantRequest request = mock(JudgementWarrantRequest.class);
+        JudgementWarrantResponse response = mockJudgementWarrantService.judgementWarrantRequest(AUTHORISATION,
+                                                                                                SERVICE_AUTHORISATION,
+                                                                                                IDAM_ID_HEADER,
+                                                                                                SDT_REQUEST_ID , request);
+        assertNotNull(response);
+        assertEquals(45678L, response.getFee());
+        assertEquals("123456", response.getWarrantNumber());
```
[the remark this answers is not in the document]
Comment (kiran-yenigala-hmcts) [reply consumers/src/unit-test/java/uk/gov/moj/sdt/cmc/consumers/client/mock/MockJudgementWarrantServiceTest.java:43]: done

File: consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/client/mock/MockJudgementWarrantService.java (hunk 4 above)
Comment (kiran-yenigala-hmcts) [reply consumers/src/main/java/uk/gov/moj/sdt/cmc/consumers/client/mock/MockJudgementWarrantService.java:19]: done

**Review by paulridout — APPROVED** [review]
Changes look OK.

**Pull request merged**